File: Otasukejuru/app/src/main/java/jp/ict/muffin/otasukejuru/HoldableViewPager.java
package jp.ict.muffin.otasukejuru;


import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class HoldableViewPager extends ViewPager {
    private boolean isSwipeHoldVar = false;

    HoldableViewPager(Context context) {
        super(context);
    }

    HoldableViewPager(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
    }

    public void setSwipeHoldVar(boolean swipeHoldVar) {
        isSwipeHoldVar = swipeHoldVar;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return !isSwipeHoldVar && super.onTouchEvent(ev);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return isSwipeHoldVar && super.onInterceptTouchEvent(ev);
    }
}

File: Otasukejuru/app/src/main/kotlin/jp/ict/muffin/otasukejuru/HoldableViewPager.kt
package jp.ict.muffin.otasukejuru

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent

/**
 * Created by mito on 2017/08/23.
 */
class HoldableViewPager: android.support.v4.view.ViewPager {
    var isSwipeHoldVar = false
    constructor(context: Context): super(context)
    constructor(context: Context, attrs: AttributeSet): super(context, attrs)
    
    fun setSwipeHold(enable: Boolean) {
        isSwipeHoldVar = enable
    }
    
    override fun onTouchEvent(ev: MotionEvent?): Boolean = !isSwipeHoldVar && super.onTouchEvent(ev)
    override fun onInterceptTouchEvent(ev: MotionEvent?): Boolean =
            !isSwipeHoldVar && super.onInterceptTouchEvent(ev)
}